#
# Patch developed in house.
# The issue was reported to the Upstream by external person: Bug #3699.
# This patch ifdef out the symbols already defined in the FIPS canister.
#
--- a/crypto/sparccpuid.S	2017-10-13 01:02:08.616856430 -0700
+++ b/crypto/sparccpuid.S	2017-10-13 01:03:23.741412970 -0700
@@ -321,6 +321,7 @@
 .type	_sparcv9_fmadd_probe,#function
 .size	_sparcv9_fmadd_probe,.-_sparcv9_fmadd_probe
 
+#if 0
 .global	_sparcv9_rdcfr
 .align	8
 _sparcv9_rdcfr:
@@ -344,6 +345,7 @@
 	.word	0x91b002a0	!random	%o0
 .type	_sparcv9_random,#function
 .size	_sparcv9_random,.-_sparcv9_vis3_probe
+#endif
 
 .global	OPENSSL_cleanse
 .align	32
--- a/crypto/sparcv9cap.c	2017-10-16 12:07:43.548063300 -0700
+++ b/crypto/sparcv9cap.c	2017-10-16 12:07:52.514879215 -0700
@@ -12,7 +12,7 @@
 #if defined(__GNUC__) && defined(__linux)
 __attribute__ ((visibility("hidden")))
 #endif
-unsigned int OPENSSL_sparcv9cap_P[2] = { SPARCV9_TICK_PRIVILEGED, 0 };
+extern unsigned int OPENSSL_sparcv9cap_P[2];
 
 int bn_mul_mont(BN_ULONG *rp, const BN_ULONG *ap, const BN_ULONG *bp,
                 const BN_ULONG *np, const BN_ULONG *n0, int num)
